nginx.tpl.conf

```
@@ -36,12 +36,37 @@ http {
     server {
         listen 80 default_server;
 
+        error_page   404  /404.html;
+
         rewrite ^/assets/white_paper.* https://github.com/jibrelnetwork/white_paper permanent;
 
         location / {
             root /app;
-            index index.html;
-            try_files $uri $uri.html $uri/ /index.html;
+            try_files $uri $uri/ =404;
+
+            location ~* \.(?:manifest|appcache|html?|xml|json)$ {
+                expires -1;
+                add_header Cache-Control "public, must-revalidate";
+            }
+
+            location ~* /img/[^/]+\.(?:jpg|jpeg|gif|png|ico|svg|webp)$ {
```

- Without leading `^` it can have collisions like `/docs/img/filename.jpg`
- Maybe it would be safer to include hash existence check here? 

```suggestion
            location ~* ^/img/[^/]+\.[a-f0-9]{8}\.(?:jpg|jpeg|gif|png|ico|svg|webp)$ {
```